src/ui.form-input/UInput.vue

```diff
@@ -166,173 +71,20 @@ const emit = defineEmits([
-   */
-  labelAlign: {
-    type: String,
-    default: getDefault(defaultConfig, UInput).labelAlign,
-  },
-
-  /**
-   * Input placeholder.
-   */
-  placeholder: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Input description.
-   */
-  description: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Error message.
-   */
-  error: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Input size.
-   * @values sm, md, lg
-   */
-  size: {
-    type: String,
-    default: getDefault(defaultConfig, UInput).size,
-  },
-
-  /**
-   * Left icon name.
-   */
-  leftIcon: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Right icon name.
-   */
-  rightIcon: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Maximum character length.
-   */
-  maxLength: {
-    type: [String, Number],
-    default: "",
-  },
-
-  /**
-   * Prevents some characters from input.
-   * You can use predefined values or own RegExp.
-   * @values symbol, string, stringAndNumber, number, integer
-   */
-  validationRule: {
-    type: [String, RegExp],
-    default: getDefault(defaultConfig, UInput).validationRule,
-    validator: (value) => Object.keys(VALIDATION_RULES_REG_EX).includes(value) || value === "",
-  },
-
-  /**
-   * Input type.
-   * @values text, number, tel, email, url, search, password
-   */
-  type: {
-    type: String,
-    default: getDefault(defaultConfig, UInput).type,
-  },
-
-  /**
-   * Set specific keyboard for mobile devices.
-   * @values text, decimal, numeric, tel, email, url, search, none
-   */
-  inputmode: {
-    type: String,
-    default: getDefault(defaultConfig, UInput).inputmode,
-  },
-
-  /**
-   * Make input read-only.
-   */
-  readonly: {
-    type: Boolean,
-    default: getDefault(defaultConfig, UInput).readonly,
-  },
-
-  /**
-   * Disable the input.
-   */
-  disabled: {
-    type: Boolean,
-    default: getDefault(defaultConfig, UInput).disabled,
-  },
-
-  /**
-   * Disable browsers autocomplete.
-   */
-  noAutocomplete: {
-    type: Boolean,
-    default: getDefault(defaultConfig, UInput).noAutocomplete,
-  },
-
-  /**
-   * Unique element id.
-   */
-  id: {
-    type: String,
-    default: "",
-  },
-
-  /**
-   * Component config object.
-   */
-  config: {
-    type: Object,
-    default: () => ({}),
-  },
-
-  /**
-   * Data-test attribute for automated testing.
-   */
-  dataTest: {
-    type: String,
-    default: "",
-  },
-});
+const VALIDATION_RULES_REG_EX = {
+  integer: /\d*/g,
+  number: /\d*\.?\d*/g,
+  string: /[a-zA-Z]+/g,
+  stringAndNumber: /[a-zA-Z0-9]+/g,
+  symbol: /\D/g,
+};
 
 const slots = useSlots();
 
 const isShownPassword = ref(false);
-const inputRef = ref(null);
-const labelComponentRef = ref(null);
-const leftSlotWrapperRef = ref(null);
+const inputRef = ref<HTMLInputElement | null>(null);
+const leftSlotWrapperRef = ref<HTMLElement | null>(null);
+const labelComponentRef = ref<{ labelElement: HTMLElement } | null>(null);
```

Comment:
![image](https://github.com/user-attachments/assets/6d1cb1f8-3a77-44ed-8998-6d14388fd040)

Because inside `labelComponentRef` (which is an object) there are two elements - label and wrapper. In setLabelPosition function we need to access labelElement particularly.